### PR TITLE
Introduce add and put function for table type

### PR DESCRIPTION
**Pull request by pcnfernando**

### CODE REVIEW

**Comment by mohanvive** [comment]
@pcnfernando Please send this PR directly to master.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java [hunk 1]
```diff
@@ -376,19 +389,13 @@ public V getData(K key) {
             return values.get(TableUtils.hash(key, new ArrayList<>()));
         }
 
-        public V putData(K key, V data) {
```
Comment (gimantha) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java:379]: Have you checked the case where the key is something we give intentionally, but the actual key is something else.
For example,
table<Person> key(name) t = table[];
t["Chiran"] = {name: "NotChiran"};
How does the above case work?

Comment (pcnfernando) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java]: This works.
But when I use an Int as key and use it for index based access I get an ArrayOutOfBound.

Below is the sample I tried.

```ballerina
type Customer record {
  readonly int id;
  string firstName;
  string lastName;
};
type CustomerTable table<Customer> key(id);

function testTableIndexAccess() returns boolean {
boolean testPassed = true;
CustomerTable custTbl = table [
{ id: 1, firstName: "Sanjiva", lastName: "Weerawarana" },
{ id: 5, firstName: "Gimantha", lastName: "Bandara" }
];

custTbl[5] = { id: 5, firstName: "Gimantha", lastName: "Bandaraaa" };

return testPassed;
}
```

Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java]: @pcnfernando can you plz get an update from the master and see if this is fixed.. I fixed a similar issue

Comment (pcnfernando) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java]: Yes, this works as expected now. Thanks

**Comment by pcnfernando** [comment]
Let's not merge this until the inconsistent issue is figured out.

**Pull request merged**